src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisStatusUpdateService.java

```diff
@@ -66,29 +71,38 @@ public void handleStatusUpdate(CourseChatJob job, PyrisChatStatusUpdateDTO statu
      *
      * @see PyrisStageState#isTerminal()
      *
-     * @param statusUpdate the status update
-     * @param job          the job to remove
+     * @param stages the stages of the status update
+     * @param job    the job to remove
      */
-    private void removeJobIfTerminated(PyrisChatStatusUpdateDTO statusUpdate, String job) {
-        var isDone = statusUpdate.stages().stream().map(PyrisStageDTO::state).allMatch(PyrisStageState::isTerminal);
+    private boolean removeJobIfTerminated(List<PyrisStageDTO> stages, String job) {
+        var isDone = stages.stream().map(PyrisStageDTO::state).allMatch(PyrisStageState::isTerminal);
         if (isDone) {
             pyrisJobService.removeJob(job);
+            return true;
         }
+        return false;
     }
 
     /**
      * Handles the status update of a lecture ingestion job and logs the results for now => will change later
-     * TODO: Update this method to handle changes beyond logging
+     *
      *
      * @param job          the job that is updated
      * @param statusUpdate the status update
      */
     public void handleStatusUpdate(IngestionWebhookJob job, PyrisLectureIngestionStatusUpdateDTO statusUpdate) {
-        statusUpdate.stages().forEach(stage -> log.info(stage.name() + ":" + stage.message()));
-        boolean isDone = statusUpdate.stages().stream().map(PyrisStageDTO::state)
-                .allMatch(state -> state == PyrisStageState.DONE || state == PyrisStageState.ERROR || state == PyrisStageState.SKIPPED);
-        if (isDone) {
-            pyrisJobService.removeJob(job.jobId());
+        if (removeJobIfTerminated(statusUpdate.stages(), job.jobId()) && statusUpdate.id().isPresent()) {
+            if (attachmentUnitRepository.findById(statusUpdate.id().get()).isPresent()) {
+                AttachmentUnit unit = attachmentUnitRepository.findById(statusUpdate.id().get()).get();
+                if (statusUpdate.stages().getLast().state() == PyrisStageState.DONE) {
+                    unit.setPyrisIngestionState(IngestionState.DONE);
+                    attachmentUnitRepository.save(unit);
+                }
+                else if (statusUpdate.stages().getLast().state() == PyrisStageState.ERROR || statusUpdate.stages().getLast().state() == PyrisStageState.SKIPPED) {
```

Comment:
Can you check these conditions before loading the attachment unit to avoid unnecessary load on the data base?

Reply:
they are already checkt in the first condition by the removeJobIfTerminated method.